src/types.ts

```diff
@@ -266,4 +311,12 @@ export class NamedType implements TypeName, Type {
     accept<T>(visitor: TypeVisitor<T>): T {
         return visitor.visitNamedType(this)
     }
+
+    equal(other: Type): boolean {
+        return (
+            other instanceof NamedType
```

Comment:
we're not checking the module and exported status. is exported part of the type? is the module part of the type? perhaps remove that from NamedType and put it closer to the module concept

Reply:
this is a bit outside the scope of this PR. clearly NamedType adds an orthogonal concept to types, but we're sort of using it as is. I'm willing to keep the inconsistency until it becomes clearer.